File: Objective-C/iOSFastlaneIntegrationUITests/iOSFastlaneIntegrationUITests.m
//
//  iOSFastlaneIntegrationUITests.m
//  iOSFastlaneIntegrationUITests
//
//  Created by John Lima on 01/05/17.
//  Copyright © 2017 John Lima. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "SnapshotHelper.h"

@interface iOSFastlaneIntegrationUITests : XCTestCase

@end

@implementation iOSFastlaneIntegrationUITests

- (void)setUp {
    [super setUp];
    
    XCUIApplication *app = [[XCUIApplication alloc] init];
    SnapshotHelper *snap = [[SnapshotHelper alloc] initWithApp:app];
    [app launch];
    [self setContinueAfterFailure:NO];
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

- (void)testGetScreenshots {
    
    XCUIApplication *app = [[XCUIApplication alloc] init];
    SnapshotHelper *snap = [[SnapshotHelper alloc] initWithApp:app];
    [app launch];
    
    XCUIElementQuery *cells = app.tables.cells;
    [[cells elementBoundByIndex:0] tap];
    [snap snapshot:@"Home" waitForLoadingIndicator:NO];
}

@end

File: Swift/iOSFastlaneIntegrationUITests/iOSFastlaneIntegrationUITests.swift
//
//  iOSFastlaneIntegrationUITests.swift
//  iOSFastlaneIntegrationUITests
//
//  Created by John Lima on 30/04/17.
//  Copyright © 2017 John Lima. All rights reserved.
//

import XCTest

class iOSFastlaneIntegrationUITests: XCTestCase {
        
    override func setUp() {
        super.setUp()
        
        let app = XCUIApplication()
        setupSnapshot(app)
        app.launch()
        
        continueAfterFailure = false
    }
    
    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }
    
    func testExample() {
        // Use recording to get started writing UI tests.
        // Use XCTAssert and related functions to verify your tests produce the correct results.
    }
    
    func testGetScreenshots() {
    
        let app = XCUIApplication()
        setupSnapshot(app)
        app.launch()
        
        snapshot("Home")
        app.tables.cells.element(boundBy: 0).tap()
    }
}
